tests/test_primod/test_ribamod_exchange_creator.py

```diff
@@ -0,0 +1,243 @@
+        da = xr.ones_like(self.conductance)
+        gridded_basin = xr.full_like(self.gridded_basin, 3)
+        gridded_basin[..., 2:] = np.nan
+        basin_index, include = exc._find_coupled_cells(
+            da, gridded_basin, self.basin_ids
+        )
+
+        # Only the first two columns are included
+        assert basin_index.size == (3 * 4 * 2)
+        # All cells are located in the second basin.
+        assert (basin_index == 1).all()
+        # Check entries per layer
+        assert include[..., :2].all()
+        assert (~include[..., 2:]).all()
+
+    def test_find_coupled_cells__some_conductance(self):
+        da = xr.ones_like(self.conductance)
+        gridded_basin = xr.full_like(self.gridded_basin, 3)
+        da[..., 2:] = np.nan
+        basin_index, include = exc._find_coupled_cells(
+            da, gridded_basin, self.basin_ids
+        )
+
+        # Only the first two columns are included
+        assert basin_index.size == (3 * 4 * 2)
+        # All cells are located in the second basin.
+        assert (basin_index == 1).all()
+        # Check entries per layer
+        assert include[..., :2].all()
+        assert (~include[..., 2:]).all()
+
+    def test_find_coupled_cells__multiple_basins(self):
+        da = xr.ones_like(self.conductance)
+        da[1:, ...] = np.nan
+        gridded_basin = xr.full_like(self.gridded_basin, 3)
+        gridded_basin[..., 2:] = 11
+        basin_index, include = exc._find_coupled_cells(
+            da, gridded_basin, self.basin_ids
+        )
+
+        assert np.array_equal(
+            basin_index, [1, 1, 3, 3, 3, 1, 1, 3, 3, 3, 1, 1, 3, 3, 3, 1, 1, 3, 3, 3]
+        )
+        assert include[0, ...].all()
+        assert (~include[1:, ...]).all()
+
+    def test_find_coupled_cells(self):
+        da = conductance()
+        gridded_basin = xr.full_like(self.gridded_basin, 3)
+        gridded_basin[:, 1] = 7
+        gridded_basin[:, 2] = 11
+        basin_index, _ = exc._find_coupled_cells(da, gridded_basin, self.basin_ids)
+        assert np.array_equal(basin_index, [1, 2, 3])
+
+
+def test_derive_boundary_index():
+    da = conductance()
+    include = da.notnull().to_numpy()
+    boundary_id = exc._derive_boundary_index(da, include)
+    assert np.array_equal(boundary_id, [0, 1, 2])
+
+    da = xr.ones_like(conductance())
+    boundary_id = exc._derive_boundary_index(da, include)
+    assert np.array_equal(boundary_id, [0, 26, 52])
+
+
+def test_derive_passive_coupling():
+    da = conductance()
+    gridded_basin = xr.full_like(da.isel(layer=0, drop=True), 7)
+    basin_ids = pd.Series([1, 3, 7, 11])
+
+    table = exc.derive_passive_coupling(da, gridded_basin, basin_ids)
+    assert isinstance(table, pd.DataFrame)
+    assert table.shape == (3, 2)
+    pd.testing.assert_frame_equal(
+        table,
+        pd.DataFrame(data={"basin_index": [2, 2, 2], "bound_index": [0, 1, 2]}),
+        check_dtype=False,  # int32 versus int64 ...
```

Comment:
Which integer to choose seems to be an art form...

Reply:
I'm somewhat hopeful the new numpy release will fix this?
https://numpy.org/devdocs/release/2.0.0-notes.html#default-integer-now-64bit-on-64bit-windows